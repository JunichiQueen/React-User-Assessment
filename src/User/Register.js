import React, { Component } from 'react';
import axios from 'axios';

export default class Register extends Component {
    constructor(){
        super();
        this.state={
            success: "",
            username: "",
            email: "",
            password: "",
            password2: ""
        }
    }

    createUser = (e) => {
        e.preventDefault();
        let newUser = {
            username: e.target[0].value,
            email: e.target[1].value,
            password: e.target[2].value,
            password2: e.target[3].value
        }
        axios
        .post("http://localhost:5000/user/createUser", newUser).then(response => {
            this.props.getAll();
            if (response.data.username){
                this.setState({
                    username: response.data.username.toString()
                })
            }
            if (response.data.email){
                    this.setState({
                        email: response.data.email.toString()
                    })
            }
            if (response.data.password){
                    this.setState({
                        password: response.data.password.toString()
                    })
            }
            if (response.data.password2){
                this.setState({
                    password2: response.data.password2.toString()
                })
            }
            if (response.data._id) {
                this.setState({
                    success: "You have created a user",
                    email: "",
                    password: ""
                })
            }

            console.log(response);
        }).catch(err => {
            console.log(err)
        });
    }
    render(){
        return(
            <div>
                <h1 className="text-center jumbotron">Register</h1>
                <form className="text-center" onSubmit={this.createUser}>
                    <p>Username</p>
                    <input></input>
                    <p style={{color: 'red'}}>{this.state.username}</p>
                    <p>Email</p>
                    <input></input>
                    <p style={{color: 'red'}}>{this.state.email}</p>
                    <p>Password</p>
                    <input></input>
                    <p style={{color: 'red'}}>{this.state.password}</p>
                    <p>Password2</p>
                    <input></input>
                    <p style={{color: 'red'}}>{this.state.password2}</p>
                    <br></br>
                    <button type="submit">Create Account</button>
                    <p style={{color: 'green'}}>{this.state.success}</p>
                    
                </form>
            </div>
        )
    }

}
