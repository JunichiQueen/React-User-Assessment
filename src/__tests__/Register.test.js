import React from 'react';
import ReactDOM from 'react-dom';
import Register from '../User/Register';
import { shallow } from "enzyme";
import renderer from 'react-test-renderer'
import "../setupTests"

it('renders without crashing', () => {
    shallow(<Register />);
});
  
it('snapshot app', () => {
    const wrapper = renderer.create(<Register />).toJSON();
    expect(wrapper).toMatchSnapshot();
});