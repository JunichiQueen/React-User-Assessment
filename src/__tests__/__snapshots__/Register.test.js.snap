// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot app 1`] = `
<div>
  <h1
    className="text-center jumbotron"
  >
    Register
  </h1>
  <form
    className="text-center"
    onSubmit={[Function]}
  >
    <p>
      Username
    </p>
    <input />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
    <p>
      Email
    </p>
    <input />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
    <p>
      Password
    </p>
    <input />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
    <p>
      Password2
    </p>
    <input />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
    <br />
    <button
      type="submit"
    >
      Create Account
    </button>
    <p
      style={
        Object {
          "color": "green",
        }
      }
    >
      
    </p>
  </form>
</div>
`;
